fuel_test/prep_env.sh

```diff
@@ -6,6 +6,58 @@ set -eu
 
 [ $DEBUG == "on" ] && set -x
 
+function create_networks {
+	local xs_host="$1"
+	ssh -qo StrictHostKeyChecking=no root@$xs_host \
+	'
+	set -eux
+	[ -z $(xe network-list name-label="'$2'" --minimal) ] && xe network-create name-label="'$2'"
+	[ -z $(xe network-list name-label="'$3'" --minimal) ] && xe network-create name-label="'$3'"
+	[ -z $(xe network-list name-label="'$4'" --minimal) ] && xe network-create name-label="'$4'"
+	echo "Network created"
+	'
+}
+
+function recreate_gateway {
+	local xs_host="$1"
+	ssh -qo StrictHostKeyChecking=no root@$xs_host \
+	'
+	set -eux
+	bridge=$(xe network-list name-label="'$2'" params=bridge minimal=true)
+	recreate_gateway_sh="/etc/udev/scripts/recreate-gateway.sh"
+	if [ ! -x $recreate_gateway_sh ]; then
+		cat > $recreate_gateway_sh << RECREATE_GATEWAY
+#!/bin/bash
+if /sbin/ip link show $bridge > /dev/null 2>&1; then
+  if !(/sbin/ip addr show $bridge | /bin/grep -q 172.16.1.1); then
+    /sbin/ip addr add dev $bridge 172.16.1.1
+  fi
+  if !(/sbin/route -n | /bin/grep -q 172.16.1.0); then
+    /sbin/route add -net 172.16.1.0 netmask 255.255.255.0 dev $bridge
+  fi
+  if !(/sbin/iptables -t nat -S | /bin/grep -q 192.168.111.0/24); then
+    /sbin/iptables -t nat -A POSTROUTING -d 192.168.111.0/24 -j RETURN
+  fi
+  if !(/sbin/iptables -t nat -S | /bin/grep -q 10.0.7.0/24); then
+    /sbin/iptables -t nat -A POSTROUTING -d 10.0.7.0/24 -j RETURN
+  fi
+  if !(/sbin/iptables -t nat -S | /bin/grep -q 10.1.7.0/24); then
+    /sbin/iptables -t nat -A POSTROUTING -d 10.1.7.0/24 -j RETURN
+  fi
+  if !(/sbin/iptables -t nat -S | /bin/grep -q 172.16.1.0/24); then
+    /sbin/iptables -t nat -A POSTROUTING -s 172.16.1.0/24 ! -d 172.16.1.0/24 -j MASQUERADE
+  fi
+fi
+RECREATE_GATEWAY
+		chmod +x $recreate_gateway_sh
+		$recreate_gateway_sh
```

Comment:
Add a comment here saying we should run the script now because we don't want to reboot to activate.